### PR TITLE
Add make module.

### LINKED ISSUE #16: Dynamic tasks
Hi, this is less of an issue and more of a "I found out how to do this and maybe someone will find it useful".

So, the idea is that it is possible to have dynamic tasks as part of a module.
When creating your own module, you can do

```lua
{
  params = ...,
  condition = ...,
  tasks = setmetatable({
      -- normal (static) commands
    }, {
    __index = function(tab, key)
      return function(module_config, _)
        -- return task configuration based on requested task 'key'...
      end
    end
  })
}
```

An example would be to wrap `make` in a module and support arbitrary targets.

```lua
{
  params = { 'cmd' },
  condition = function()
    return require('plenary.path'):new('Makefile'):exists()
  end,
  tasks = setmetatable({}, {
    __index = function(_, target)
      return function(module_config, _)
        return {
          cmd = module_config.cmd,
          args = module_config.args and module_config.args[target] or { target },
          -- Close quickfix on success
          after_success = function() vim.cmd.cclose() end
        }
      end
    end
  })
}
```

With this, calling `Task start make install` will run `make install` (i.e., the task is mapped to the `make` target).
This behavior could also be customized if we have the following in our setup

```lua
{
  default_params = {
    make = {
      cmd = "make",
      args = {
        build = {},
        nuke = { "clean" }
      } 
  }
}
```

Now
- `Task start make build` -> `make` 
- `Task start make nuke` -> `make clean`

By adding a metatable to `default_params.make.args`, it could be possible to add dynamic task parameters for dynamic tasks.

Now, the example is not very useful unless one wants to use `make` with the same interface used for other modules, but you get the gist of it.

### PR DESCRIPTION
**Pull request by undyamon**
Supports dynamic tasks, mapped to make targets by default.

Closes #16

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -90,6 +92,31 @@ require('tasks').setup({
 
 Cargo module doesn't have a `target` param which specific to CMake because `cargo run` automatically pick the binary. If there is multiple binaries, you can set which one you want to run using `--bin` or `--project` in step 2 as you do in CLI.
 
+### GNU Make
+
+1. Open a Make project.
+1. Run a Make target `<target>` with `:Task start make <target>`.
```
Comment (Shatur) [inline README.md:98]: I know that Markdown automatically numerates lists, but above I used manual numeration. Could you do it as above just for consistency?
```suggestion
2. Run a Make target `<target>` with `:Task start make <target>`.
```

File: README.md [hunk 2]
```diff
@@ -27,6 +27,8 @@ Tasks in this plugin are provided by modules that implement functionality for a
 - [CMake](https://cmake.org) via [cmake-file-api](https://cmake.org/cmake/help/latest/manual/cmake-file-api.7.html#codemodel-version-2).
 - [Cargo](https://doc.rust-lang.org/cargo).
 
+- [GNU Make](https://www.gnu.org/software/make/)
```
Comment (Shatur) [inline README.md:30]: Could you remove a newline before `GNU Make`?

**Review by Shatur — CHANGES_REQUESTED** [review]
Thanks! A few suggestions and I will merge it.

**Comment by Shatur** [comment]
Could you also fix formatting, please?

**Comment by undyamon** [comment]
Sure what do you use to format Lua code?

**Comment by Shatur** [comment]
Stylua, but you could just apply diff from the failed job to avoid installing a new tool.

**Comment by Shatur** [comment]
Looks like still fails due to an extra newline.

**Comment by undyamon** [comment]
bruh

**Pull request merged**